code/go/0chain.net/smartcontract/storagesc/validator.go

```diff
@@ -19,12 +22,13 @@ func (sc *StorageSmartContract) addValidator(t *transaction.Transaction, input [
 	}
 	newValidator.ID = t.ClientID
 	newValidator.PublicKey = t.PublicKey
+	newValidator.ProviderType = spenum.Validator
```

Comment:
We don't need to add this filed to differ the different provider type, instead it's the `GetKey()` method that matters. So the format of key can be `${SCAddress}:${provider_type}:${ID}` instead of `provider:${ID}`.   Otherwise, if blobber and validator got the same ID, getting node from the same `key` will cause problem.

Reply:
The idea is to prevent blobbers and validators from having the same id. Is there a reason why we want to allow blobbers and validators to have the same id.

Reply:
@Sriep I agree with @peterlimg, we can simply generate key in a different way, for example
```go
validator.GetKey(ADDRESS)
```
can be 
```go

func (sn *ValidationNode) GetKey(_ string) datastore.Key {
	return provider.GetKey(sn.ID, spenum.Validator)
}
```
and then as peter wrote ``key := ${SCAddress}:${provider_type}:${ID} ``
as we fully control writings to the MPT it should be safe

Reply:
can we remove this field then?

Reply:
What field?

`key := ${SCAddress}:${provider_type}:${ID}` This does not do what we want as different providers will have different keys due to ` ${SCAddress}:${provider_type}:` part being different. 

For example, if a blobber and a miner have the same id we want the MPT key to be the same, so a conflict is immediately flagged.

Reply:
```
 Otherwise, if blobber and validator got the same ID, getting node from the same key will cause problem.
```
This should be impossible as this PR will prevent blobbers and validators from ever having the same id.

The whole point of this PR is to create this conflict so that we can prevent different providers from having the same id. When we create a second provider with the same id, the conflict is immediately detected and the second provider cannot be created.